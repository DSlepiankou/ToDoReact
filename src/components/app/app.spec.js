import React from "react";
import App from "./app";
import {render}  from '@testing-library/react';
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

const testStore = mockStore({
    toDoList: {
        elements: [{}, {}, {}]
    }
})

it('App component contains nested components', ()=> {
    
    const todoCounter = render(
        <Provider store={testStore}>
            <App />
        </Provider>)
    
    expect(todoCounter).toMatchSnapshot();
})