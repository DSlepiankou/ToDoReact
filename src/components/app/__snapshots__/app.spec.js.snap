// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component contains nested components 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app"
      >
        <div
          class="to-do-list"
        >
          <h1
            class="header-text"
          >
            ToDo List
          </h1>
          <form
            class="item-add-form"
          >
            <input
              class="form-control"
              placeholder="Type here what you want todo"
              type="text"
              value=""
            />
          </form>
          <ul
            class="list-group todo-list"
          >
            <li
              class="list-group-item"
            >
              <span
                class="todo-list-item"
              >
                <div>
                  <input
                    class="checkbox-circle"
                    type="checkbox"
                  />
                </div>
                <button
                  class="float-right delete-btn"
                  type="button"
                >
                  <i
                    class="bi bi-trash"
                  />
                </button>
              </span>
            </li>
            <li
              class="list-group-item"
            >
              <span
                class="todo-list-item"
              >
                <div>
                  <input
                    class="checkbox-circle"
                    type="checkbox"
                  />
                </div>
                <button
                  class="float-right delete-btn"
                  type="button"
                >
                  <i
                    class="bi bi-trash"
                  />
                </button>
              </span>
            </li>
            <li
              class="list-group-item"
            >
              <span
                class="todo-list-item"
              >
                <div>
                  <input
                    class="checkbox-circle"
                    type="checkbox"
                  />
                </div>
                <button
                  class="float-right delete-btn"
                  type="button"
                >
                  <i
                    class="bi bi-trash"
                  />
                </button>
              </span>
            </li>
          </ul>
          <span
            class="item-status-panel"
          >
            <div
              class="counter"
            >
              3
               more to do
            </div>
            <div
              class="btn-group-filter"
            >
              <button
                class="butn outline"
                type="button"
              >
                All
              </button>
              <button
                class="butn outline"
                type="button"
              >
                Active
              </button>
              <button
                class="butn outline"
                type="button"
              >
                Done
              </button>
            </div>
            <button
              class="del-button"
            >
              Delete completed
            </button>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app"
    >
      <div
        class="to-do-list"
      >
        <h1
          class="header-text"
        >
          ToDo List
        </h1>
        <form
          class="item-add-form"
        >
          <input
            class="form-control"
            placeholder="Type here what you want todo"
            type="text"
            value=""
          />
        </form>
        <ul
          class="list-group todo-list"
        >
          <li
            class="list-group-item"
          >
            <span
              class="todo-list-item"
            >
              <div>
                <input
                  class="checkbox-circle"
                  type="checkbox"
                />
              </div>
              <button
                class="float-right delete-btn"
                type="button"
              >
                <i
                  class="bi bi-trash"
                />
              </button>
            </span>
          </li>
          <li
            class="list-group-item"
          >
            <span
              class="todo-list-item"
            >
              <div>
                <input
                  class="checkbox-circle"
                  type="checkbox"
                />
              </div>
              <button
                class="float-right delete-btn"
                type="button"
              >
                <i
                  class="bi bi-trash"
                />
              </button>
            </span>
          </li>
          <li
            class="list-group-item"
          >
            <span
              class="todo-list-item"
            >
              <div>
                <input
                  class="checkbox-circle"
                  type="checkbox"
                />
              </div>
              <button
                class="float-right delete-btn"
                type="button"
              >
                <i
                  class="bi bi-trash"
                />
              </button>
            </span>
          </li>
        </ul>
        <span
          class="item-status-panel"
        >
          <div
            class="counter"
          >
            3
             more to do
          </div>
          <div
            class="btn-group-filter"
          >
            <button
              class="butn outline"
              type="button"
            >
              All
            </button>
            <button
              class="butn outline"
              type="button"
            >
              Active
            </button>
            <button
              class="butn outline"
              type="button"
            >
              Done
            </button>
          </div>
          <button
            class="del-button"
          >
            Delete completed
          </button>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
